grails-app/controllers/org/pih/warehouse/JsonController.groovy

```diff
@@ -1754,31 +1753,25 @@ class JsonController {
         Organization destinationParty = Organization.get(params.destinationPartyId)
         List productSuppliers = []
         if (product && supplier) {
-            productSuppliers = ProductSupplier.createCriteria().list {
-                eq("product", product)
-                eq("supplier", supplier)
-                or {
-                    isEmpty("productSupplierPreferences")
-                    productSupplierPreferences {
-                        and {
-                            eq("destinationParty", destinationParty)
-                            preferenceType {
-                                not {
-                                    eq("validationCode", ValidationCode.HIDE)
-                                }
-                            }
-                        }
-                    }
-                }
-            }
+            productSuppliers = dataService.executeQuery(
+                    """
+                        SELECT ps.id, ps.code, ps.supplier_code, ps.name, ps.manufacturer_code, ps.manufacturer_id from product_supplier ps
+                        LEFT OUTER JOIN product_supplier_preference default_preference ON default_preference.product_supplier_id = ps.id AND default_preference.destination_party_id IS NULL
+                        LEFT OUTER JOIN  product_supplier_preference preference ON preference.product_supplier_id = ps.id AND preference.destination_party_id = '${destinationParty.id}'
+                        WHERE product_id = '${product.id}'
```

Comment:
Add parameter for ${product.id} and ${supplier.id} (something like: `:productId`, `:supplierId`) and pass it in the parameters map.

Reply:
+1